src/particle.h

```diff
@@ -37,11 +39,17 @@ class Particle : protected Pointers {
     double rottemp[3];      // rotational temperature(s)
     double vibtemp[MAXVIBMODE];   // vibrational temperature(s)
     double vibrel[MAXVIBMODE];    // inverse vibrational relaxation number(s)
+    double electemp[MAXELECSTATE];  // electronic temperatures for each electronic state
+    double elecrel[MAXSPECIES][MAXELECSTATE];   // inverse electronic collision numbers
```

Comment:
`double elecrel[MAXSPECIES][MAXELECSTATE]` is pretty big and will use up significant stack memory (12*16*8 bytes) and will bloat the Species struct when no electronic modes are defined. It may be better to define a new struct that is only allocated when electronic modes are defined. I'm open to Steve's opinion on this though.

Reply:
That seems reasonable to me. Are you thinking essentially leave only `nelecstate` plus a pointer to the new struct in the `Species` struct, then we need only allocate that struct for the species where `nelecstate > 0`? Also, by delaying the allocation of that struct until slightly later (after we parse the electronic mode input file) we could allocate the vectors to only be as long as needed rather than guessing a maximum size. The parsing fits most naturally in `particle.cpp`, but would it be preferable to move the struct definition and the related methods to `fix_elecmode`? A similar method could be used to reduce the size of `elecrel`.

Reply:
@stanmoore1 I am working on this and most of the edits are straight forward. Essentially I made another struct that contains all the relevant information and only allocate it for the species that have electronic data defined. One difficulty I am running into is that during parsing in `particle.cpp`, I don't think I can trust `nspecies`. Specifically for that relaxation array where I want to store values for each species in the simulation and the following vector of bools. Would it be preferable to initialize them at some length and error if we need more space (i.e. using `MAXSPECIES`) or add logic to `grow_species` to also touch these vectors/arrays?

Reply:
Thanks for pointing that out. Since I am now allocating the memory after parsing the ElecFile it should be easy to remove those constants.

Reply:
@zseckert sounds good, after you push your changes I will port them to Kokkos

Reply:
@stanmoore1 I just pushed an update that dynamically allocates only the memory needed for the electronic mode. The MAXSPECIES and MAXELECSTATE constants are still floating around in the ElecFile struct that is used during parsing, which I will hopefully get to tomorrow, but all the memory that hasn't been released by the start of the simulation is hopefully in its final form. Let me know your thoughts. I still need to do a thorough search for dynamic array in stack memory.

Reply:
@stanmoore1 I have fixed a couple more dynamic arrays on the stack as well as modified the electronic state parsing to remove the constant sized arrays/vectors. The code for this is quite a bit messier with all the smalloc's and sfree's, but it's simple enough that I have confidence in it, and I have tested it all on my suite of examples that show the same results. I have now done everything that is on my list, unless I have overlooked something. Let me know what you need from me, but I think this is ready for you again.